scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala

```diff
@@ -406,16 +406,17 @@ class FormatOps(
     }
   }
 
-  def templateCurly(template: Template): Option[Token] = {
-    template.tokens.find(x => x.is[T.LeftBrace] && owners(x) == template)
-  }
-
-  def templateDerivesOrCurly(template: Template): Option[Token] = {
-    template.tokens.collectFirst {
-      case t: T.LeftBrace if owners(t) eq template => t
-      case t @ soft.KwDerives() if owners(t) eq template => tokens(t, -1).left
-    }
-  }
+  def templateCurly(template: Template): Option[Token] =
+    getStartOfTemplateBody(template)
+      .map(x => prevNonComment(tokens(x, -1)).left)
```

Comment:
avoids iteration over all tokens, finds it exactly where it should be. handles `colon at end of line` as well.